src/main/java/com/uber/cadence/workflow/Workflow.java

```diff
@@ -720,6 +723,32 @@ public static Scope getMetricsScope() {
     return WorkflowInternal.getMetricsScope();
   }
 
+  /**
+   * Get logger to use inside workflow. Logs in replay mode are omitted unless {@param
+   * enableLoggingInReplay} is set to true in {@link WorkerOptions} when a worker starts up.
+   *
+   * @param clazz class name to appear in logging.
+   * @return logger to use in workflow logic.
+   */
+  public static Logger getLogger(Class<?> clazz) {
+    Logger logger = LoggerFactory.getLogger(clazz);
+    return new ReplayAwareLogger(
+        logger, WorkflowInternal::isReplaying, WorkflowInternal::isLoggingEnabledInReplay);
+  }
+
+  /**
+   * Get logger to use inside workflow. Logs in replay mode are omitted unless {@param
+   * enableLoggingInReplay} is set to true in {@link WorkerOptions} when a worker starts up.
+   *
+   * @param name name to appear in logging.
+   * @return logger to use in workflow logic.
+   */
+  public static Logger getLogger(String name) {
+    Logger logger = LoggerFactory.getLogger(name);
+    return new ReplayAwareLogger(
```

Comment:
I would make ReplayAwareLogger non public and move this code to WorkflowInternal.